media/Jsontracer/detail.js

```diff
@@ -0,0 +1,207 @@
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// This file referenced the result of
+// https://github.com/catapult-project/catapult/tree/444aba89e1c30edf348c611a9df79e2376178ba8/tracing
+
+const timeUnit = {
+  ms: 10 ** -3,
+  us: 1,
+  ns: 10 ** 3,
+};
+
+export function renderSingleDetail(data) {
+  const detailSection = document.querySelector('.detail-container section');
+  const ul = document.createElement('ul');
+  ul.className = 'detail';
+
+  Object.keys(data).map((key) => {
+    // invisible backgroundColor and pk
+    if (key === 'backgroundColor' || key === 'pk') {
+      return;
+    }
+    const li = document.createElement('li');
+
+    if (key === 'args') {
+      // show args in list
+      li.innerText = 'args';
+      li.className = 'args';
+      li.addEventListener('click', () => {
+        li.classList.toggle('fold');
+      });
+      li.append(renderArgs(data[key]));
+    } else if (key === 'ts' || key === 'dur') {
+      // show time with displayTimeUnit
+      const setData = document.querySelector('.set-data');
+      const displayTimeUnit = setData.dataset['displayTimeUnit'];
+      li.innerText = `${key} : ${parseInt(data[key] * timeUnit[displayTimeUnit] * 1000) / 1000} ${
+          displayTimeUnit}`;
+    } else {
+      // show others
+      li.innerText = `${key} : ${data[key]}`;
+    }
+
+    // append DOM
+    ul.append(li);
+    detailSection.append(ul);
+  });
+}
+
+function renderArgs(args) {
+  const ul = document.createElement('ul');
+  args.split('.#/#.').forEach(element => {
+    const li = document.createElement('li');
+    li.innerText = element;
+    ul.append(li);
+  });
+  return ul;
+}
+
+export function renderMultipleDetail() {
+  makeTable();
+  refinedSelectedOp();
+}
+
+function makeTable() {
+  const detailSection = document.querySelector('.detail-container section');
+  const table = document.createElement('table');
+  table.className = 'detail';
+
+  const thead = document.createElement('thead');
+  const tr = document.createElement('tr');
+
+  const thName = document.createElement('th');
+  thName.innerText = 'name';
+
+  const thWall = document.createElement('th');
+  thWall.innerText = 'Wall Duration';
+
+  const thAverage = document.createElement('th');
+  thAverage.innerText = 'Average Wall Duration';
+
+  const thOccurrences = document.createElement('th');
+  thOccurrences.innerText = 'Occurrences';
+
+  tr.append(thName, thWall, thAverage, thOccurrences);
+  thead.append(tr);
+  table.append(thead);
+  detailSection.append(table);
+}
+
+function refinedSelectedOp() {
+  const selectedOpList = document.querySelectorAll('.selected-op');
+  const refinedOP = [];
+  const refinedOPDict = {};
+  let idx = 0;
+
+  selectedOpList.forEach(element => {
+    const name = element.dataset.name;
+    const ts = element.dataset.ts * 1;
```

Comment:
Why multiply 1?

Reply:
This is to change the string to num. Automatic transformation was used.

Reply:
I'll change it to `Number` for readability.